app/models.py

```diff
@@ -464,11 +464,15 @@ def from_service(service):
         )
 
     def copy(self):
+        data = self.data.copy()
+        name = data.get('serviceName', '')
+        if len(name) <= 95:
+            data['serviceName'] = u"{} copy".format(name)
```

Comment:
That was my first idea too, but multiple "Copy of Copy of" look (somewhat) worse than " copy copy", and the appended text is what both Windows and OS X do.